crypto/src/test/java/org/web3j/crypto/Bip32Test.java

```diff
@@ -0,0 +1,167 @@
+                "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL",
+                new int[]{0, 2147483647 | HARDENED_BIT, 1, 2147483646 | HARDENED_BIT});
+
+        // Chain m/0/2147483647H/1/2147483646H/2
+        testGenerated("fffcf9f6f3f0edeae7e4e1dedbd8d5d2cfccc9c6c3c0bdbab7b4b1aeaba8a5a29f9c999693908d8a8784817e7b7875726f6c696663605d5a5754514e4b484542",
+                "xprvA2nrNbFZABcdryreWet9Ea4LvTJcGsqrMzxHx98MMrotbir7yrKCEXw7nadnHM8Dq38EGfSh6dqA9QWTyefMLEcBYJUuekgW4BYPJcr9E7j",
+                "xpub6FnCn6nSzZAw5Tw7cgR9bi15UV96gLZhjDstkXXxvCLsUXBGXPdSnLFbdpq8p9HmGsApME5hQTZ3emM2rnY5agb9rXpVGyy3bdW6EEgAtqt",
+                new int[]{0, 2147483647 | HARDENED_BIT, 1, 2147483646 | HARDENED_BIT, 2});
+    }
+
+    @SuppressWarnings("checkstyle:LineLength")
+    @Test
+    public void deriveKeyPairVector3() {
+        // Chain m
+        testGenerated("4b381541583be4423346c643850da4b320e46a87ae3d2a4e6da11eba819cd4acba45d239319ac14f863b8d5ab5a0d0c64d2e8a1e7d1457df2e5a3c51c73235be",
+                "xprv9s21ZrQH143K25QhxbucbDDuQ4naNntJRi4KUfWT7xo4EKsHt2QJDu7KXp1A3u7Bi1j8ph3EGsZ9Xvz9dGuVrtHHs7pXeTzjuxBrCmmhgC6",
+                "xpub661MyMwAqRbcEZVB4dScxMAdx6d4nFc9nvyvH3v4gJL378CSRZiYmhRoP7mBy6gSPSCYk6SzXPTf3ND1cZAceL7SfJ1Z3GC8vBgp2epUt13",
+                null);
+
+        // Chain m/0H
+        testGenerated("4b381541583be4423346c643850da4b320e46a87ae3d2a4e6da11eba819cd4acba45d239319ac14f863b8d5ab5a0d0c64d2e8a1e7d1457df2e5a3c51c73235be",
+                "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L",
+                "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y",
+                new int[]{0 | HARDENED_BIT});
+    }
+
+    private void testGenerated(String seed, String expectedPriv, String expectedPub, int[] path) {
+        Bip32ECKeyPair pair = Bip32ECKeyPair.generateKeyPair(hexToByte(seed));
+        assertNotNull(pair);
+
+        pair = Bip32ECKeyPair.deriveKeyPair(pair, path);
+        assertNotNull(pair);
+
+        assertEquals(expectedPriv, Base58.encode(addChecksum(serializePrivate(pair))));
+        assertEquals(expectedPub, Base58.encode(addChecksum(serializePublic(pair))));
+    }
+
+    static byte[] addChecksum(byte[] input) {
+        int inputLength = input.length;
+        byte[] checksummed = new byte[inputLength + 4];
+        System.arraycopy(input, 0, checksummed, 0, inputLength);
+        byte[] checksum = hashTwice(input);
+        System.arraycopy(checksum, 0, checksummed, inputLength, 4);
+        return checksummed;
+    }
+
+    static byte[] serializePublic(Bip32ECKeyPair pair) {
+        return serialize(pair, 0x0488B21E, true);
+    }
+
+    static byte[] serializePrivate(Bip32ECKeyPair pair) {
+        return serialize(pair, 0x0488ADE4, false);
+    }
+
+    private byte[] hexToByte(String encoded) {
```

Comment:
This function is already defined - https://github.com/web3j/web3j/blob/master/utils/src/main/java/org/web3j/utils/Numeric.java#L174

Reply:
Should Bip32Test.java be named Bip32ECKeyPairTest.java?

Reply:
fixed, let me know if I should also rename the file